### PR TITLE
integrate response content-type's expected value

### PR DESCRIPTION
**Pull request by moonbingbing**
fix this issue https://github.com/openresty/openresty/issues/92

### CODE REVIEW

File: src/ngx_http_lua_headers.c [hunk 1]
```diff
@@ -603,12 +603,19 @@ ngx_http_lua_ngx_header_get(lua_State *L)
...
+    ngx_int_t                    rc;
 
     r = ngx_http_lua_get_req(L);
     if (r == NULL) {
         return luaL_error(L, "no request object found");
     }
 
+    ctx = ngx_http_get_module_ctx(r, ngx_http_lua_module);
+    if (ctx == NULL) {
+        return luaL_error(L, "no ctx");
```
Comment (doujiang24) [inline src/ngx_http_lua_headers.c]: better use `no ctx found` here :)

File: src/ngx_http_lua_headers.c [hunk 2]
```diff
@@ -640,6 +647,17 @@ ngx_http_lua_ngx_header_get(lua_State *L)
...
 
+    if (!ctx->headers_set) {
+        rc = ngx_http_lua_set_content_type(r);
+        if (rc != NGX_OK) {
+            return luaL_error(L,
+                              "failed to set default content type: %d",
+                              (int) rc);
+        }
+
+        ctx->headers_set = 1;
```
Comment (doujiang24) [inline src/ngx_http_lua_headers.c]: I'm afraid this is wrong here, better remove this line, like this case

```lua
local content_type = ngx.header.content_type
ngx.header.foo = "foo"
```

File: t/016-resp-header.t [hunk 3]
```diff
@@ -1441,3 +1441,49 @@ Content-Type: ; blah
...
+=== TEST 70:always return the matched content-type
+--- config
+    location /set/ {
+        default_type "image/png";
+        content_by_lua_block {
+            ngx.say(ngx.header["content-type"])
+            ngx.say(ngx.header["content-type"])
+        }
+        header_filter_by_lua_block {
+        }
```
Comment (doujiang24) [inline t/016-resp-header.t]: useless code here?

**Comment by moonbingbing** [comment]
@doujiang24 thanks for your helpful code review. I will fix them ASAP.

**Comment by moonbingbing** [comment]
I will also fix ` ngx.resp.get_headers` :)

File: src/ngx_http_lua_headers.c [hunk 4]
```diff
@@ -475,6 +477,21 @@ ngx_http_lua_ngx_resp_get_headers(lua_State *L)
...
+    }
+
+    if (!ctx->headers_set) {
+        rc = ngx_http_lua_set_content_type(r);
+        if (rc != NGX_OK) {
+            return luaL_error(L,
+                              "failed to set default content type: %d",
+                              (int) rc);
+        }
+        ctx->headers_set = 1;
```
Comment (agentzh) [inline src/ngx_http_lua_headers.c]: Style: need a blank line after `}`.

File: t/016-resp-header.t [hunk 5]
```diff
@@ -1441,3 +1441,71 @@ Content-Type: ; blah
...
+--- response_headers
+Content-Type: image/png
+--- response_body
+image/png
+--- no_error_log
+[error]
+
+
+
+=== TEST 70:always return the matched content-type
```
Comment (agentzh) [inline t/016-resp-header.t]: Style: need a space after the colon (`:`).

File: src/ngx_http_lua_headers.c [hunk 6]
```diff
@@ -475,6 +477,22 @@ ngx_http_lua_ngx_resp_get_headers(lua_State *L)
...
+    }
+
+    if (!ctx->headers_set) {
+        rc = ngx_http_lua_set_content_type(r);
+        if (rc != NGX_OK) {
+            return luaL_error(L,
+                              "failed to set default content type: %d",
+                              (int) rc);
+        }
+
```
Comment (agentzh) [inline src/ngx_http_lua_headers.c]: This line has trailing spaces. Better use the `ngx-releng` tool to check such things. For this case, `ngx-releng` reports the following error (the error message was actually in red):

```
src/ngx_http_lua_headers.c:
found unnecessary tail space
492:
```

**Comment by agentzh** [comment]
@moonbingbing I've fixed the last coding style nit and also the commit log messages :) Thanks!

**Pull request closed without merging**